library/src/main/java/com/rengwuxian/materialedittext/TypefaceCache.java

```diff
@@ -0,0 +1,20 @@
+package com.rengwuxian.materialedittext;
+
+import android.content.res.AssetManager;
+import android.graphics.Typeface;
+
+import java.util.Hashtable;
+
+public class TypefaceCache {
+  private static final Hashtable<String, Typeface> FONT_CACHE = new Hashtable<>();
+
+  public static Typeface get(String name, AssetManager assets) {
+    synchronized (FONT_CACHE) {
```

Comment:
yes, you are right about the weakreference but I would prefer to reload it than having it in memory if not used.